### PR TITLE
Disallow editing of readonly entity children

### PR DESCRIPTION
**Pull request by ianeli1**
Add checks for both inline styling and element deletion to disallow any changes to *children* of readonly entity elements.

Currently, children elements of readonly elements can be styled by changing the color and font size, as well as using `Ctrl + X` to cut text. 

Given that both `deleteSelectedContent` and `applyInlineStyle` work by calling callbacks on an array of HTML elements, we can verify if those elements/nodes are ~~children of entities~~ editable before passing them to the callback, disallowing said changes to these readonly entities.

An additional check was added to prevent nodes inside a readonly container to be merged together.

### CODE REVIEW

File: packages/roosterjs-editor-api/lib/utils/applyInlineStyle.ts [hunk 1]
```diff
@@ -27,7 +42,7 @@ export default function applyInlineStyle(
                 (getTagOfNode(node.firstChild) == 'BR' && !node.firstChild.nextSibling));
         if (isEmptySpan) {
             editor.addUndoSnapshot();
-            callback(node as HTMLElement);
+            entitySafeCallback(node as HTMLElement);
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-api/lib/utils/applyInlineStyle.ts:45]: I think we only need to check if this node is read only (node.isContentEditable), no need to check entity.

Comment (ianeli1) [reply packages/roosterjs-editor-api/lib/utils/applyInlineStyle.ts:45]: Do you mean this for all the cases, or just the empty `<span>`?

Comment (JiuqingSong) [reply packages/roosterjs-editor-api/lib/utils/applyInlineStyle.ts:45]: for all the cases

Comment (ianeli1) [reply packages/roosterjs-editor-api/lib/utils/applyInlineStyle.ts]: Yup, you were right 😁

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**